
version: '3.7'

services:
  web:
    build:
      context: .
      network: host
      dockerfile: Dockerfile
    container_name: web
    image: web:lastest
    command: python manage.py runserver 0.0.0.0:8000
    restart: always
    ports:
      - "8000:8000"
    env_file:
      - .env
    networks:
      - app
    volumes:
      - .:/django  
    depends_on:
      - db

  celery:
    build: .
    restart: on-failure
    command: celery -A config worker -l INFO
    volumes:
      - .:/django
    networks:
      - app
    depends_on:
      - web
      - redis

  redis:
    image: redis
    container_name: redis
    restart: on-failure
    ports:
      - "6379:6379"
    networks:
      - app
      
  db:
    image: postgres
    restart: always
    environment:
      - POSTGRES_DB=movie
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    volumes:
      - postgres_db:/var/lib/postgresql/data
    networks:
      - app

  pgadmin:
    image: dpage/pgadmin4
    environment:
      - PGADMIN_DEFAULT_EMAIL=test@gmail.com
      - PGADMIN_DEFAULT_PASSWORD=test
    networks:
      - app

networks:
  app:

volumes:
  postgres_db:
